src/paperqa/core.py

```diff
@@ -276,14 +276,14 @@ async def _map_fxn_summary(  # noqa: PLR0912
 
         llm_results.append(llm_result)
         context = llm_result.text or ""
-        try:
-            result_data = parser(context) if parser else {}
```

Comment:
This little refactor removes the extra `if parser else {}`, making the code slightly less stateful